crates/engine/tree/src/tree/mod.rs

```diff
@@ -1787,18 +1802,18 @@ where
             let tip = chain_update.tip().header.clone();
             self.on_canonical_chain_update(chain_update);
 
+            if let Some(attr) = attrs {
+                let updated = self.process_payload_attributes(attr, &tip, state);
+                return Ok(TreeOutcome::new(updated))
+            }
```

Comment:
do we want to move this before the following checks?

Reply:
sorry, yeah we definitely need to keep the previous location